### DISCUSSION TITLE
Is the probe output information stored in the restart file?

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#746] [Q&A] [answered]
Say I'm running a 200 ns simulation and output the probe information every 20 ns. If I restart from the restart file at 195 ns, does it still "remember" the particles passing through the probe from 180 ns to 195 ns?

**Comment by Status-Mirror** [comment]
Hi @wnguyen1312,

The particle probes only store particle-hits since the last SDF file output. I think you'd only keep particles from 195ns to 200ns in the next dump.

I'm not sure if you can find the missing particles from 180-195ns in the restart dump or not. Maybe you can check to see if there is probe data in this file. If there's no probe data in the restart file, then the code wouldn't be able to continue this probe data anyway after the restart.

Hope this helps,
Stuart

**Reply by cooluser26** [reply]
Hi Stuart,

It looks like the probe data is not stored in the restart. Is the best way to fix this is just to output the the probe data a lot more frequently?

Thanks!

**Reply by Status-Mirror** [reply]
I'd just try to make the probe-dumps occur on the same time-steps as the restart dumps. The problem you mention above wouldn't happen if you did your restart on 200ns instead of 195ns, because the probes from 180ns-200ns would be logged in the 200ns dump. You could then restart at 200ns, and the next probe dump would have 200ns-220ns as expected.

**Reply by cooluser26** [reply]
The issue is that my simulation takes 3-4 days, but the cluster’s wall time is only 12 hours, so I’m using `touch STOP` to restart. However, the particle probe data doesn’t save in restarts, leading to data gaps. Is there any trick to sync probe dumps with restart steps to avoid this?

**Reply by Status-Mirror** [reply, marked as answer]
Oh I see. We have a similar issue with some big cluster simulations that we've been running. Maybe it's worth looking into this instead of just accepting gaps in the data. I'm going to mark this as an issue that we can hopefully create a solution for. Maybe we should make those forced restart-dumps also dump the probe data.

We don't have much spare developer time to push this change through quick, so in the meantime I think dumping probes more regularly would be the easier solution.

**Reply by Status-Mirror** [reply]
This has been fixed in the latest EPOCH release - restart dumps will now always dump particle probe information.

Thanks for pointing this one out!
Stuart
